cmd/snapshot-create.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/Layr-Labs/sidecar/internal/logger"
+	"github.com/spf13/cobra"
+)
+
+var (
+	snapshotCreateToPath string
+	snapshotFidelity     string
+	s3Path               string
+	compressSnapshot     bool
+)
+
+var snapshotCreateCmd = &cobra.Command{
+	Use:   "snapshot-create",
+	Short: "Create a snapshot of the database",
+	Long: `Create a snapshot of the database.
+
+Currently available fidelity levels:
+- archive (default): includes chain data, EigenModel state, rewards, and staker-operator table data.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Initialize logger (in a real scenario, you'd likely pass in a config)
+		l, _ := logger.NewLogger(&logger.LoggerConfig{Debug: true})
```

Comment:
Look at this file for inspiration on how to set things up properly:

https://github.com/Layr-Labs/sidecar/blob/master/cmd/database.go#L35-L109